woocommerce/woocommerce-ios
# Shipping Labels: New package details selection

Part of #4599 
This PR depends on #4961 so please make sure that PR is reviewed and merged first.

# Description
This PR adds selection logic for the new package details screen. To reuse the logic between the old and the new screens, the selection logic is extracted into a separate view model.

# Changes
- Adds new view model for package list and copies the logic from `ShippingLabelPackageDetailsViewModel`.
- Updates the correct view model for `ShippingLabelPackageList` and `ShippingLabelPackageSelection` views.
- Removes package list and selection logic from `ShippingLablePackageDetailsViewModel` and adds package list view model.
- Adds package list view model to `ShippingLabelPackageItemViewModel` to handle package switching.
- Handles configuring package total weight in  `ShippingLabelPackageItemViewModel`.
- Updates `ShippingLabelPackageItemViewModel` initializer with new block for notifying the packages form of new package.
- Updates `ShippingLabelPackagesFormViewModel` to update selected package list.
- Updates unit tests for both package item and package details view models.

# Demo
![package-selection](https://user-images.githubusercontent.com/5533851/132680791-0e7677db-9456-4afa-ad86-3fdf56fac3de.gif)

# Testing

1. Make sure that your test store has installed WCShip plugin and configured packages in WPAdmin > WooCommerce > Settings > Shipping > WooCommerce Shipping.
2. Open an order detail that is eligible for the shipping label.
3. Configure Ship To and Ship From.
4. Tap Continue on Package Details row. Switch between packages, notice that the correct title and weight of the package is displayed. Edit the weight to an arbitrary number and switch packages, notice that the weight is no longer updated.
5. Turn off multi-package feature flag, repeat steps 2, 3 and 4. Verify that everything still works correctly.

Update release notes:
